linchpin/InventoryFilters/AWSInventory.py

```diff
@@ -38,29 +38,36 @@ def get_host_data(self, topo, cfgs):
         """
 
         host_data = {}
+        if res['resource_type'] != 'aws_ec2_res':
+            return host_data
         var_data = cfgs.get('aws', {})
         if var_data is None:
             var_data = {}
-        for group in topo.get('aws_ec2_res', []):
-            for instance in group['instances']:
-                host = self.get_hostname(instance, var_data,
-                                         self.DEFAULT_HOSTNAMES)
-                hostname_var = host[0]
-                hostname = host[1]
-                if '__IP__' not in var_data.keys():
-                    var_data['__IP__'] = hostname_var
-                host_data[hostname] = {}
-                self.set_config_values(host_data[hostname], instance, var_data)
+        for instance in res['instances']:
+            host = self.get_hostname(instance, var_data,
+                                     self.DEFAULT_HOSTNAMES)
+            hostname_var = host[0]
+            hostname = host[1]
+            if '__IP__' not in var_data.keys():
+                var_data['__IP__'] = hostname_var
+            host_data[hostname] = {}
+            self.set_config_values(host_data[hostname], instance, var_data)
         return host_data
 
     def get_host_ips(self, host_data):
-        return host_data.keys()
+        if host_data:
+            return host_data.keys()
+        else:
+            return []
 
     def get_inventory(self, topo, layout, config):
-        if len(topo['aws_ec2_res']) == 0:
-            return ""
-        host_data = self.get_host_data(topo, config)
-        inven_hosts = self.get_host_ips(host_data)
+        host_data = []
+        inven_hosts = []
+        for host in topo:
```

Comment:
nit: I would call this "res" here instead of "host".  That matches what your get_host_data method is doing above.  it's pretty minor but helps plebs like me who don't know the code intimately. (and if you do change it, change it for all the providers)

Reply:
Do you mean `res_data` instead of `host_data`? Will do

Reply:
that sounds good too.  But I was talking about 
```
for host in topo:
```
you call it res (I assume resource?) in the get_res_data method.  so res_data and for res in topo: would be clearer to me.  But like I said.  really minor nit.

Reply:
I see